### PR TITLE
dev/core#1858 Prevent Duplicate contact records being created and har…

### PR DESCRIPTION
**Pull request by seamuslee001**
…d fatal errors when creating a CMS user account from the contact action by ensuring that the contact_id field is correctly passed through for dedupe matching

Overview
----------------------------------------
Reproduction steps are on the lab ticket https://lab.civicrm.org/dev/core/-/issues/1858 but the key aspect of the reproduction steps is having an unsupervised dedupe rule for individuals that requires not just a match on email address to work

Before
----------------------------------------
DB error / duplicate contact created

After
----------------------------------------
No DB Error / no duplicate contact record created

ping @demeritcowboy @eileenmcnaughton @mlutfy @KarinG @kcristiano

### CODE REVIEW

**Comment by demeritcowboy** [comment]
* General standards
    * [`r-explain`] __Ok__
      * It's explored in the ticket that Unsupervised is involved because it calls a function that in its own words "can be called standalone". Otherwise it's confusing why Unsupervised matters.
      * Also the problem doesn't affect drupal 7 because drupal 8 always calls it and it's a different function getting called than drupal 7.
    * [`r-user`] __PASS__
    * [`r-doc`] __PASS__
    * [`r-run`] __PASS__
* Developer standards
    * [`r-tech`] __Could use a code comment as to why__
      * In light of the above and that it's a bit hard to follow what's going on and as discussed in the ticket, this seems a safer option, but it's going to be even more confusing to someone later. It could say something like `// The "Create User Record" action for contacts uses a different field name`.
    * [`r-code`] __Ok__
      * To be super-safe it should probably be `if (empty($params['contact_id']) && !empty($params['contactID'])) {`, but I don't know a path where that would come up where both would already be set *AND* they'd be different and then the "real" one would get overwritten with the "wrong" one.
    * [`r-maint`] __Undecided__
    * [`r-test`] __PASS__

**Comment by mattwire** [comment]
@seamuslee001 Can you add a code comment per @demeritcowboy suggestion?

**Comment by seamuslee001** [comment]
done

**Comment by demeritcowboy** [comment]
Great, thanks.

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I was gonna merge this & then I read the comment :

"case from the back off contact task
        // That this gets passed onto the duplicate function. "

I'm not sure that makes sense?

**Comment by seamuslee001** [comment]
@eileenmcnaughton have clarified the comment now hopefully

**Comment by eileenmcnaughton** [comment]
thanks!
